### PR TITLE
[Java][Native] added WithHttpInfo method for native-sync

### PR DESCRIPTION
**Pull request by bgong-mdsol**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/config/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
This PR added  **WithHttpInfo** method for **native** only, it includes the changes below:
- updated JavaClientCodegen
- updated the template of api
- updated the template of api-doc
- regenerated Petstore samples for native/asyncNative

@wing328 @cbornet @jeff9finger @bbdouglas @sreeshas @jfiala @lukoyanov @karismann @Zomzog @lwlee2608 @bkabrda @jfeltesse-mdsol @jcarres-mdsol

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 1]
```diff
@@ -404,6 +404,9 @@ public void processOpts() {
         } else if (NATIVE.equals(getLibrary())) {
             setJava8Mode(true);
             additionalProperties.put("java8", "true");
+            if (!asyncNative) {
+                supportingFiles.add(new SupportingFile("ApiResponse.mustache", invokerFolder, "ApiResponse.java"));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: Would it be possible to also use ApiResponse in async operations to make it consistent?

Comment (bgong-mdsol) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: @wing328 This `ApiResponse` is used by `WithHttpInfo` method which is implemented for `sync` operation only. Currently `async` operation doesn't need `ApiResponse`. I can include `ApiResponse` in `async` for future use, does anyone need `WithHttpInfo` method  for `async` operation?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: > does anyone need WithHttpInfo method for async operation?

For consistency, I suggest we implement it as well if it's not a lot of work (we're implemented the same in other client generators)

Comment (jfeltesse-mdsol) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: My concern is, do we know whether people are actually using it? I'd hate we spend time on this when we might have a case of YAGNI.
If it is indeed needed, would it be ok to do that in a separate PR to facilitate the scheduling of work?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: Valid concern. That's why I asked if it's a lot of work.

Feel free to do it later with another PR instead.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:408]: And one more thing to add is that usually the `sync` operations in other clients simply call the `async` version of the operation and wait for completion.

**Pull request merged**